web-common/src/Tooltip/State.purs

```diff
@@ -0,0 +1,125 @@
+module Tooltip.State (component, tooltip) where
+
+import Prelude
+import Control.Bind (bindFlipped)
+import Control.Monad.Maybe.Trans (MaybeT(..), runMaybeT)
+import Data.Foldable (for_)
+import Data.Lens (assign, set, use)
+import Data.Maybe (Maybe(..))
+import Data.Symbol (SProxy(..))
+import Effect (Effect)
+import Effect.Aff.Class (class MonadAff)
+import Halogen (Component, HalogenM, Slot, get, getHTMLElementRef, liftEffect, mkComponent, modify_)
+import Halogen as H
+import Halogen.HTML (ComponentHTML, HTML, slot)
+import Halogen.Query.EventSource (eventListenerEventSource)
+import Popper (OffsetOption(..), PaddingOption(..), Placement, PositioningStrategy(..), arrow, createPopper, defaultModifiers, defaultPreventOverflow, destroyPopper, forceUpdate, offset, pAll, preventOverflow)
+import Tooltip.Lenses (_active, _mPopperInstance, _message, _placement)
+import Tooltip.Types (Action(..), Input, RefferenceId(..), State, arrowRef, tooltipRef)
+import Tooltip.View (render)
+import Web.DOM.NonElementParentNode (getElementById)
+import Web.Event.Event (EventType(..))
+import Web.HTML (HTMLElement, window)
+import Web.HTML.HTMLDocument (toNonElementParentNode)
+import Web.HTML.HTMLElement as HTMLElement
+import Web.HTML.Window (document)
+
+_tooltipSlot :: SProxy "tooltipSlot"
+_tooltipSlot = SProxy
+
+tooltip ::
+  forall slots m action.
+  MonadAff m =>
+  String ->
+  RefferenceId ->
+  Placement ->
+  ComponentHTML action ( tooltipSlot :: forall query. Slot query Void RefferenceId | slots ) m
+tooltip message reference placement = slot _tooltipSlot reference component { message, reference, placement } absurd
```

Comment:
if/when we update to Halogen 6, I think we can use `slot_` here.....

Reply:
yes, I first looked at the documentation and tried to use `slot_` only to realize it was not available in v5 😅

**when** we update 💪